src/Razor/Razor.Runtime/src/RazorRazorPageBase.cs

```diff
@@ -0,0 +1,505 @@
+            }
+
+            if (SectionWriters.ContainsKey(name))
+            {
+                // TODO throw new InvalidOperationException(Resources.FormatSectionAlreadyDefined(name));
+                throw new InvalidOperationException($"Format Section Already Defined ({name})");
+            }
+            SectionWriters[name] = section;
+        }
+
+
+        /// <summary>
+        /// Writes the specified <paramref name="value"/> with HTML encoding to <see cref="Output"/>.
+        /// </summary>
+        /// <param name="value">The <see cref="object"/> to write.</param>
+        public virtual void Write(object value)
+        {
+            if (value == null || value == HtmlString.Empty)
+            {
+                return;
+            }
+
+            var writer = Output;
+            var encoder = HtmlEncoder;
+            if (value is IHtmlContent htmlContent)
+            {
+                // TODO boffered writing in impl
+                htmlContent.WriteTo(writer, encoder);
+                return;
+            }
+
+            Write(value.ToString());
+        }
+
+        /// <summary>
+        /// Writes the specified <paramref name="value"/> with HTML encoding to <see cref="Output"/>.
+        /// </summary>
+        /// <param name="value">The <see cref="string"/> to write.</param>
+        public virtual void Write(string value)
+        {
+            var writer = Output;
+            var encoder = HtmlEncoder;
+            if (!string.IsNullOrEmpty(value))
+            {
+                // Perf: Encode right away instead of writing it character-by-character.
+                // character-by-character isn't efficient when using a writer backed by a ViewBuffer.
+                var encoded = encoder.Encode(value);
+                writer.Write(encoded);
+            }
+        }
+
+        /// <summary>
+        /// Writes the specified <paramref name="value"/> without HTML encoding to <see cref="Output"/>.
+        /// </summary>
+        /// <param name="value">The <see cref="object"/> to write.</param>
+        public virtual void WriteLiteral(object value)
+        {
+            if (value == null)
+            {
+                return;
+            }
+
+            WriteLiteral(value.ToString());
+        }
+
+        /// <summary>
+        /// Writes the specified <paramref name="value"/> without HTML encoding to <see cref="Output"/>.
+        /// </summary>
+        /// <param name="value">The <see cref="string"/> to write.</param>
+        public virtual void WriteLiteral(string value)
+        {
+            if (!string.IsNullOrEmpty(value))
+            {
+                Output.Write(value);
+            }
+        }
+
+        public virtual void BeginWriteAttribute(
+            string name,
+            string prefix,
+            int prefixOffset,
+            string suffix,
+            int suffixOffset,
+            int attributeValuesCount)
```

Comment:
@pranavkm @NTaylorMullen - any objection to simplifying this? We can add a new overload, update the compiler and then reduce this proliferation

@m0sa - FYI these int values are used for some diagnostics things that aren't supported anymore.

Reply:
Simplify!!!!! 👍